core/deployment/src/main/java/io/quarkus/deployment/recording/FieldsHelper.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.deployment.recording;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
+final class FieldsHelper {
+
+    private final Map<String, Field> fields;
+
+    public FieldsHelper(final Class<?> aClass) {
+        final Field[] declaredFields = aClass.getDeclaredFields();
+        this.fields = new HashMap<>(declaredFields.length);
```

Comment:
Not sure it's a good idea to do that, your `Map` will be resized always given the default `loadFactor` of `0.75`.

In HV, we had a util who correctly created a Map of the right size for all the maps initialized at bootstrap: https://github.com/hibernate/hibernate-validator/blob/729cd018599d046aa5db1dbe35700521dd596c4e/engine/src/main/java/org/hibernate/validator/internal/util/CollectionHelper.java#L142-L147 .

And sure, it's a micro optimization but in your case, you might create more harm than none by always sizing the map a bit too small.

Reply:
what are you suggesting, would you prefer I simply use `new HashMap<>()`; ?

FWIW I know about the optimisation in Validator and we had a similar thing in ORM, but it's not taking into account the likelyhood of hash collisions.. that's why there is a load factor in the hashmap implementation: it's hard to estimate perfectly. One could argue that the default loadfactor is a bit low..  My rule of thumb is currently that as long as we resize just a couple times it's fine, so I like passing the estimated size (when we have it) so to have it right at least within order of magnitude.  I guess since in this case we expect the lifespan of the map to be short one should rather overestimate: I'll  shift it right once so to double its size?

Reply:
BTW just checked, on recent JVMs the HashMap implementation is different and it looks like that this is no longer a problem.

Might be time to simplify Validator as well :)